### ISSUE TITLE
check stanc md5sum against fixed+released string

### ISSUE THREAD

**Issue by wds15** [stan-dev/cmdstan#755]
#### Summary:
Since we are now downloading **binary** files to the user machine we should really ensure that what we downloaded is the right thing. This is in order to protect against man-in-the-middle attacks.

At the very least the md5sums of the stanc compiler should be documented somewhere.

#### Description:
We should store in the cmdstan repository the expected md5 hashes of the stanc binaries and automatically verify that we really downloaded these files.

There could even be an alternative download program available which does this automatically, but this is probably not part of RTools (and I forgot the name of this wget alternative).

#### Reproducible Steps:
NA

#### Current Output:
NA

#### Expected Output:
Describe what you expect the output to be. Knowing the correct behavior is also very useful.

#### Additional Information:
Provide any additional information here.

#### Current Version:
v2.20.0

**Comment by seantalts** [comment]
We're downloading over `https` so the only man-in-the-middle attacks that are possible are those when someone else controls your local certificate store; i.e. in a large corporation that hijacks https like the one I used to work for, or possibly folks in China. In other words, I don't think this is a real attack vector unless I'm thinking about it wrong. I used to do research in computer security but that was a decade ago and I haven't kept up with all of the advances.
